File: backend/cografi-nesne-takip/src/main/java/com/kutalmis/cografi_nesne_takip/entity/PowerSource.java
package com.kutalmis.cografi_nesne_takip.entity;

public enum PowerSource {
    GRID("Şebeke"),
    SOLAR("Güneş Paneli");

    private final String displayName;

    PowerSource(String displayName) {
        this.displayName = displayName;
    }

    public String getDisplayName() {
        return displayName;
    }
}

File: frontend/lib/models/enum/power_source.dart
enum PowerSource {
  grid('Şebeke', 'GRID'),
  solar('Güneş Paneli', 'SOLAR');

  final String displayName;
  final String apiValue;

  const PowerSource(this.displayName, this.apiValue);

  static PowerSource fromApiValue(String value) {
    return PowerSource.values.firstWhere(
      (e) => e.apiValue == value,
      orElse: () => throw ArgumentError('Bilinmeyen Cisim Türü: $value'),
    );
  }
}
